launch/llmctl/src/main.rs

```diff
@@ -303,147 +246,104 @@ struct ModelRow {
-            Some(mt) => println!(
-                "Listing {} models in the public namespace: {}",
-                mt.as_str(),
-                namespace
-            ),
-            None => println!("Listing all models in the public namespace: {}", namespace),
+            Some(mt) => println!("Listing {} models in namespace: {}", mt.as_str(), namespace),
+            None => println!("Listing all models in namespace: {}", namespace),
         }
         println!("{}", table);
     }
     Ok(())
 }
 
-async fn remove_model(
-    distributed: &DistributedRuntime,
-    namespace: String,
-    model_type: ModelType,
-    name: &str,
-) -> Result<()> {
-    let component = distributed.namespace(&namespace)?.component("http")?;
-    let prefix = format!(
-        "{}/models/{}/{}",
-        component.etcd_root(),
-        model_type.as_str(),
-        name
+async fn remove_model(distributed: &DistributedRuntime, model_name: &str) -> Result<()> {
+    // We have to do this manually because normally the etcd lease system does it for us
+    let watcher = ModelWatcher::new(
+        distributed.clone(),
+        Arc::new(ModelManager::new()),
+        RouterMode::Random,
     );
-
-    log::debug!("deleting key: {}", prefix);
-
-    // get the kvs from etcd
-    let mut kv_client = distributed
-        .etcd_client()
-        .expect("llmctl is only useful with dynamic workers")
-        .etcd_client()
-        .kv_client();
-    match kv_client.delete(prefix.as_bytes(), None).await {
-        Ok(_response) => {
-            println!(
-                "{} model {} removed from the public namespace: {}",
-                model_type.as_str(),
-                name,
-                namespace
-            );
-        }
-        Err(e) => {
-            log::error!("Error removing model {}: {}", name, e);
-        }
+    let Some(etcd_client) = distributed.etcd_client() else {
+        anyhow::bail!("llmctl is only useful with dynamic workers");
+    };
+    let active_instances = watcher.entries_for_model(model_name).await?;
+    for entry in active_instances {
+        let network_name = ModelNetworkName::from_entry(&entry, 0);
+        tracing::debug!("deleting key: {network_name}");
+        etcd_client
+            .kv_delete(network_name.to_string(), None)
+            .await?;
     }
+
     Ok(())
 }
+
+fn endpoint_from_name(
+    distributed: &DistributedRuntime,
+    namespace: &str,
+    endpoint_name: &str,
+) -> anyhow::Result<Endpoint> {
+    let parts: Vec<&str> = endpoint_name.split('.').collect();
+
+    if parts.len() < 2 {
+        anyhow::bail!("Endpoint name '{}' is too short. Format should be 'component.endpoint' or 'namespace.component.endpoint'", endpoint_name);
+    } else if parts.len() > 3 {
+        anyhow::bail!("Endpoint name '{}' is too long. Format should be 'component.endpoint' or 'namespace.component.endpoint'", endpoint_name);
+    }
+
+    // TODO previous version sometime hardcoded this to "http", so maybe adjust
```

Comment:
I don't quite understand the TODO, can you clarify? Is it working as-is with an example that uses http+llmctl instead of dynamo-run for ingress?

I think the hard-coded http went hand in hand with a default component name of "http" in the http binary

Reply:
The add part requires you to call it like this: `llmctl add http model_name namespace.component.endpoint`.

It was using "component" from the cmd line for one of the places we need a component, and hard coding "http" for the other. They should match so I used "component" for both, but maybe we'll need to use "http" for both. Let's see.

Reply:
It turns out it doesn't matter. `http` doesn't use the component/etc, it listens for everything.

Reply:
Thanks for checking on it!